vendors/infineon/secure_elements/pkcs11/iot_pkcs11_trustx.c

```diff
@@ -0,0 +1,3122 @@
+ * file deviates from the FreeRTOS style standard for some function names and
+ * data types in order to maintain compliance with the PKCS #11 standard.
+ */
+
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "semphr.h"
+#include "iot_pkcs11_config.h"
+#include "iot_crypto.h"
+#include "iot_pkcs11.h"
+#include "iot_pkcs11_pal.h"
+
+/* Credential includes. */
+#include "aws_clientcredential.h"
+#include "aws_clientcredential_keys.h"
+#include "iot_default_root_certificates.h"
+
+/* OPTIGA(TM) Trust X Includes */
+#include "optiga/optiga_crypt.h"
+#include "optiga/optiga_util.h"
+#include "optiga/pal/pal_os_event.h"
+#include "optiga/ifx_i2c/ifx_i2c_config.h"
+#include "ecdsa_utils.h"
+
+#if ( pkcs11configOTA_SUPPORTED == 1 )
+    #include "aws_ota_codesigner_certificate.h"
+#endif
+
+/* C runtime includes. */
+#include <stdio.h>
+#include <string.h>
+#include <limits.h>
+
+#define PKCS11_PRINT( X )            vLoggingPrintf X
+#define PKCS11_WARNING_PRINT( X )    /* vLoggingPrintf X */
+
+#define pkcs11NO_OPERATION            ( ( CK_MECHANISM_TYPE ) 0xFFFFFFFFF )
+
+/* The size of the buffer malloc'ed for the exported public key in C_GenerateKeyPair */
+#define pkcs11KEY_GEN_MAX_DER_SIZE    200
+
+#define MAX_PUBLIC_KEY_SIZE           100
+
+#undef keyCLIENT_CERTIFICATE_PEM
+#undef keyCLIENT_PRIVATE_KEY_PEM
+
+char keyCLIENT_CERTIFICATE_PEM[1124];
+
+#define keyCLIENT_PRIVATE_KEY_PEM                   \
+"-----BEGIN EC PRIVATE KEY-----\n" \
+"MHcCAQEEIH84O6hf145Awun9lH6GbesjSmwJ5w1ODaYhp7xLZRA3oAoGCCqGSM49\n" \
+"AwEHoUQDQgAEb5ERgUCa0019j5JWdEiH0JyldJF9cv/MW2Lx94n9HVJ9FiQcooFi\n" \
+"KegulIrIuCLE1a7zsv8Izs/6gY9rxOdHHA==\n" \
+"-----END EC PRIVATE KEY-----\n"
+
+
+typedef struct P11Object_t
+{
+    CK_OBJECT_HANDLE xHandle;
+    CK_BYTE xLabel[ pkcs11configMAX_LABEL_LENGTH + 1 ]; /* Plus 1 for the null terminator. */
+} P11Object_t;
+
+typedef struct P11ObjectList_t
+{
+    SemaphoreHandle_t xMutex; /* Mutex that protects write operations to the xObjects array. */
+    P11Object_t xObjects[ pkcs11configMAX_NUM_OBJECTS ];
+} P11ObjectList_t;
+
+/* PKCS #11 Object */
+typedef struct P11Struct_t
+{
+    CK_BBOOL xIsInitialized;
+    P11ObjectList_t xObjectList;
+    uint16_t xCertOid;
+    uint16_t xPrivKeyOid;
+} P11Struct_t, * P11Context_t;
+
+static P11Struct_t xP11Context;
+
+typedef struct OptigaSha256Ctx_t
+{
+	uint8_t pubHashCtxBuf [130];
+	optiga_hash_context_t xHashCtx;
+}OptigaSha256Ctx_t;
+
+/**
+ * @brief Session structure.
+ */
+typedef struct P11Session
+{
+    CK_ULONG ulState;
+    CK_BBOOL xOpened;
+    CK_MECHANISM_TYPE xOperationInProgress;
+    CK_BBOOL xFindObjectInit;
+    CK_BBOOL xFindObjectComplete;
+    CK_BYTE * pxFindObjectLabel; /* Pointer to the label for the search in progress. Should be NULL if no search in progress. */
+    uint8_t xFindObjectLabelLength;
+    CK_MECHANISM_TYPE xVerifyMechanism;
+    SemaphoreHandle_t xVerifyMutex; /* Protects the verification key from being modified while in use. */
```

Comment:
The underlying host library may already provide this assurance, making these sign/verify mutexes redundant.

Reply:
Agree, somehow I missed this.